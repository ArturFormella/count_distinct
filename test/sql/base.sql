\set ECHO none
BEGIN;

\i sql/count_distinct--1.3.3.sql

\set ECHO all

-- int
SELECT count_distinct(x::int) FROM generate_series(1,1000) s(x);
SELECT count_distinct(mod(x,10)::int) FROM generate_series(1,1000) s(x);

-- bigint
SELECT count_distinct(x::bigint) FROM generate_series(1,1000) s(x);
SELECT count_distinct(mod(x,10)::bigint) FROM generate_series(1,1000) s(x);

-- timestamp
select count_distinct(now()::timestamp + (x || ' days')::interval) from generate_series(1,1000) s(x);
select count_distinct(now() + (x || ' days')::interval) from generate_series(1,1000) s(x);

-- bool
select count_distinct(x::bool) from generate_series(0,1000) s(x);

-- int2
select count_distinct(x::int2) from generate_series(1,1000) s(x);

ROLLBACK;
