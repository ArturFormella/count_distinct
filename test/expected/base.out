BEGIN;
CREATE EXTENSION count_distinct;
-- int
SELECT count_distinct(x::int) FROM generate_series(1,1000) s(x);
 count_distinct 
----------------
           1000
(1 row)

SELECT count_distinct(mod(x,10)::int) FROM generate_series(1,1000) s(x);
 count_distinct 
----------------
             10
(1 row)

-- bigint
SELECT count_distinct(x::bigint) FROM generate_series(1,1000) s(x);
 count_distinct 
----------------
           1000
(1 row)

SELECT count_distinct(mod(x,10)::bigint) FROM generate_series(1,1000) s(x);
 count_distinct 
----------------
             10
(1 row)

-- timestamp
select count_distinct(now()::timestamp + (x || ' days')::interval) from generate_series(1,1000) s(x);
 count_distinct 
----------------
           1000
(1 row)

select count_distinct(now() + (x || ' days')::interval) from generate_series(1,1000) s(x);
 count_distinct 
----------------
           1000
(1 row)

-- bool
select count_distinct(x::bool) from generate_series(0,1000) s(x);
 count_distinct 
----------------
              2
(1 row)

-- int2
select count_distinct(x::int2) from generate_series(1,1000) s(x);
 count_distinct 
----------------
           1000
(1 row)

ROLLBACK;
